src/main/java/org/opensearch/knn/index/codec/KNN990Codec/UnitTestCodec.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.codec.KNN990Codec;
+
+import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.apache.lucene.codecs.perfield.PerFieldKnnVectorsFormat;
+import org.opensearch.knn.index.codec.KNNCodecVersion;
+
+/**
+ * This codec is for testing. The reason for putting this codec here is SPI is only scanning the src folder and not
```

Comment:
What is SPI?

Reply:
SPI is service provider interface. This is used to create the object of a class by using name of the classes. This is how Lucene creates the Codecs without actually calling new on those classes.